advanced/wallets/react-wallet-v2/src/utils/EIP155WalletUtil.ts

```diff
@@ -4,29 +4,97 @@ import { getWalletAddressFromParams } from './HelperUtil'
 
 export let wallet1: EIP155Lib
 export let wallet2: EIP155Lib
-export let eip155Wallets: Record<string, EIP155Lib>
-export let eip155Addresses: string[]
+export let eip155Wallets: Record<string, EIP155Lib> = {}
+export let eip155Addresses: string[] = []
 
 let address1: string
 let address2: string
 
+/**
+ * Fetch E2E credentials (async, called before wallet init)
+ *
+ * Flow:
+ * 1. Pre-encrypt mnemonic with scripts/encrypt-mnemonic.js
+ * 2. Store encrypted value in E2E_ENCRYPTED_MNEMONIC secret
+ * 3. Navigate to wallet with ?e2e_encrypted=<encrypted_value>
+ * 4. This function decrypts via /api/e2e/decrypt and stores in localStorage
+ * 5. createOrRestoreEIP155Wallet() reads from localStorage
+ */
+export async function fetchE2ECredentials(): Promise<boolean> {
+  if (typeof window === 'undefined') {
+    return false
+  }
+
+  const urlParams = new URLSearchParams(window.location.search)
+  let encrypted = urlParams.get('e2e_encrypted')
+
+  // Fix URL encoding issues: spaces should be '+' in base64
+  if (encrypted) {
+    encrypted = encrypted.replace(/ /g, '+')
+  }
+
+  if (!encrypted) {
+    return false
+  }
+
+  // Clear any existing wallet to force using E2E credentials
+  localStorage.removeItem('EIP155_MNEMONIC_1')
```

Comment:
```suggestion
```
not sure if step is needed. If I pass `privateKey` via the url params, the wallet should use that but it shouldn't delete existing data imo